zebra-consensus/src/transaction.rs

```diff
@@ -717,6 +717,22 @@ where
         if let Some(orchard_shielded_data) = orchard_shielded_data {
             for authorized_action in orchard_shielded_data.actions.iter().cloned() {
                 let (action, spend_auth_sig) = authorized_action.into_parts();
+
+                // Consensus rule: The proof 𝜋 MUST be valid given a primary
+                // input (cv, rtOrchard, nf, rk, cm𝑥, enableSpends, enableOutputs)
+                //
+                // https://zips.z.cash/protocol/protocol.pdf#actiondesc
+                //
+                // Queue the verification of the Halo2 proof for each Action
+                // description while adding the resulting future to our
+                // collection of async checks that (at a minimum) must pass for
+                // the transaction to verify.
+                async_checks.push(
+                    primitives::halo2::VERIFIER
+                        .clone()
+                        .oneshot(primitives::halo2::Item::from(orchard_shielded_data)),
+                );
+
```

Comment:
You can remove this line: https://github.com/ZcashFoundation/zebra/blob/e0d9463a3554b1e52a7a8aeb6cb8408f9f513d63/zebra-consensus/src/transaction.rs#L509